openmls/src/group/tests/test_external_commit_validation.rs

```diff
@@ -869,3 +869,84 @@ fn test_pure_ciphertest(ciphersuite: Ciphersuite, backend: &impl OpenMlsCryptoPr
     // Would fail if handshake message processing did not distinguish external messages
     assert!(alice_group.parse_message(message.into(), backend).is_ok());
 }
+
+// ValSem248: External Commit must not include any proposals by reference
+#[apply(ciphersuites_and_backends)]
+fn test_valsem248(ciphersuite: Ciphersuite, backend: &impl OpenMlsCryptoProvider) {
+    // Test with MlsPlaintext
+    let ECValidationTestSetup {
+        mut alice_group,
+        bob_credential_bundle,
+        mut plaintext,
+        original_plaintext,
+    } = validation_test_setup(PURE_PLAINTEXT_WIRE_FORMAT_POLICY, ciphersuite, backend);
+
+    let mut content = if let MlsContentBody::Commit(commit) = plaintext.content() {
+        commit.clone()
+    } else {
+        panic!("Unexpected content type.");
+    };
+
+    // Add an Add proposal by reference
+    let bob_key_package = generate_key_package_bundle(
+        &[ciphersuite],
+        bob_credential_bundle.credential(),
+        vec![],
+        backend,
+    )
+    .unwrap();
+
+    alice_group
+        .propose_add_member(backend, &bob_key_package)
+        .unwrap();
```

Comment:
What does this do?

Reply:
This creates a proposal, effectively adding it to the internal ProposalQueue. That way I can reference it when I make a commit with a proposal reference.

Reply:
Right, but that's not great. The goal is to have a mechanism that reliably prevents External Commits from being processed if they reference proposals. In the test you first create such an invalid Commit (which is fine) but then you also manipulate your target so that it gives back the error you want. 
I think the detection can and should be done much earlier for this particular validation step.

Reply:
But when should it be if not when processing a commit containing such invalid reference? I also didn't understand why the target is being manipulated... The only way to create a reference is when the referenced proposal is already in the queue, otherwise adding this validation is pointless because the reference itself is not valid.

Reply:
Well, the stricter option would be not to allow External Commits that cover proposals by reference, regardless of whether the proposal is valid or not. You could check that on a purely syntactic level, once you have established what kind of a Commit it is (I think the sender type is what determines that).
Your check is implemented at a much later point in time, after the proposals have been processed into a queue already, and references have been correlated with standalone proposals.
I'm not saying that your check is completely wrong, but out of the two options, it is the less strict one.

Reply:
I see. I'll try to implement according to your suggestion

Reply:
I moved it to right before the queue is created. Because of this `valsem244` doesn't make sense anymore, so I removed it and renamed this new validation with the same name, so there's no skip between the ids. 

I see that the `valsem246` was removed from the MD file doc but it is still being validated in the code... In some of the conflicts resolution I made it wrong and it was back there... Should I remove or leave it? Isn't the idea to have all validations documented there?

Reply:
> I moved it to right before the queue is created. Because of this `valsem244` doesn't make sense anymore, so I removed it and renamed this new validation with the same name, so there's no skip between the ids.

Good thinking with repurposing valsem244, obviously it's outdated now. 

> I see that the `valsem246` was removed from the MD file doc but it is still being validated in the code... In some of the conflicts resolution I made it wrong and it was back there... Should I remove or leave it? Isn't the idea to have all validations documented there?

Where is valsem246 removed? I still see it in the the [message_validation.md](https://github.com/openmls/openmls/blob/main/book/src/message_validation.md?plain=1#L94)

Reply:
> Where is valsem246 removed? I still see it in the the [message_validation.md](https://github.com/openmls/openmls/blob/main/book/src/message_validation.md?plain=1#L94)

My mistake. The diff is showing as "readded" that's why I was confused.
![image](https://user-images.githubusercontent.com/1080789/205244471-5bba6015-4536-4b6b-92d2-6b65882c1e6a.png)